src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -195,7 +195,7 @@ protected EC2Cloud(String name, boolean useInstanceProfileForCredentials, String
         this.roleSessionName = roleSessionName;
         this.credentialsId = Util.fixEmpty(credentialsId);
         this.sshKeysCredentialsId = Util.fixEmpty(sshKeysCredentialsId);
-        if (this.sshKeysCredentialsId == null) {
+        if (this.sshKeysCredentialsId != null && this.sshKeysCredentialsId.isBlank()) {
             this.sshKeysCredentialsId = null;
```

Comment:
Remember that this will have no effect on clouds loaded from disk (`$JENKINS_HOME/config.xml`), only new saved clouds. To really enforce null vs. nonblank, you would need to either use `readResolve`, or edit the getter and then consistently use that instead of the field.

Reply:
if i implement this in `readResolve` does that make the above code redundant?

Reply:
No, because `readResolve` is not called for newly created/resaved clouds; that goes through the constructor.